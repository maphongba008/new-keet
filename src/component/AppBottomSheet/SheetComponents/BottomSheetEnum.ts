enum BottomSheetEnum {
  MessageOptionsSheet,
  ConfirmDialog,
  ConfirmBlockDialog,
  ConfirmInappropriateMessageDialog,
  RecoverWalletSheet,
  ParticipantsSheet,
  OptionsSheet,
  CallStarted,
  RoomActionSheet,
  NetworkBottomSheet,
  ChatEventOptionsSheet,
  PermissionRequired,
  Dialog,
  RoomInviteType,
  ClearAllFiles,
  FeedbackFormSheet,
  EmojiSourceSheet,
  InviteTypeOptionsSheet,
  RoomAvatarBottomSheet,
  ChannelRoomTips,
  IdentityBackupConfirmSheet,
  CleanDeviceBottomSheet,
  TurnOffPasscodeSheet,
  ChatLinkConfirmationSheet,
  RoomOptionsStats,
  CommunityContacts,
  RoomAdminReset,
  DeviceDetailsBottomSheet,
  DMRequest,
  ChatEventDeleteConfirmSheet,
  WalletSettings,
  WalletDelete,
  RoleChangeConfirm,
  ListOptionsSheet,
  CallAutoEndPopup,
}

export default BottomSheetEnum
