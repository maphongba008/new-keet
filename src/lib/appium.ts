import { ViewProps } from 'react-native'
import _includes from 'lodash/includes'

type NamedEnum<Key extends string> = { [v in Key]: v }

function inferGeneric<T extends string>(t: NamedEnum<T>) {
  return t
}

export const APPIUM_IDs = inferGeneric({
  back_button: 'back_button',
  // LobbyScreen
  lobby_btn_join_room: 'lobby_btn_join_room',
  lobby_btn_create_room: 'lobby_btn_create_room',
  lobby_btn_create_chat: 'lobby_btn_create_chat',
  lobby_btn_create_broadcast: 'lobby_btn_create_broadcast',
  lobby_btn_create_back: 'lobby_btn_create_back',
  lobby_input_room_name: 'lobby_input_room_name',
  lobby_input_room_description: 'lobby_input_room_description',
  lobby_btn_paste: 'lobby_btn_paste',
  lobby_btn_qr: 'lobby_btn_qr',
  lobby_btn_submit: 'lobby_btn_submit',
  lobby_btn_add_room: 'lobby_btn_add_room',
  lobby_tap_to_call: 'lobby_tap_to_call',
  lobby_input_search: 'lobby_input_search',
  lobby_btn_clear_search: 'lobby_btn_clear_search',
  lobby_btn_rooms: 'lobby_btn_rooms',
  lobby_btn_profile: 'lobby_btn_profile',
  lobby_dm: 'lobby_dm',
  lobby_filter: 'lobby_filter',
  lobby_selected_filter: 'lobby_selected_filter',
  lobby_joining_rooms: 'lobby_joining_rooms',
  lobby_swipable_btn_mute: 'lobby_swipable_btn_mute',
  lobby_swipable_btn_leave: 'lobby_swipable_btn_leave',
  lobby_bottomsheet_btn_leave: 'lobby_bottomsheet_btn_leave',
  // CreateBroadcastRoomScreen
  broadcast_accept_admin: 'broadcast_accept_admin',
  broadcast_btn_create: 'broadcast_btn_create',
  // RoomScreen
  room_call_join: 'room_call_join',
  room_call_mute: 'room_call_mute',
  room_call_video: 'room_call_video',
  room_call_speaker: 'room_call_speaker',
  room_call_end: 'room_call_end',
  room_btn_attach: 'room_btn_attach',
  room_btn_send: 'room_btn_send',
  room_input_msg: 'room_input_msg',
  room_attach_doc: 'room_attach_doc',
  room_attach_media: 'room_attach_media',
  room_capture_video: 'room_capture_video',
  room_capture_img: 'room_capture_img',
  room_btn_edit: 'room_btn_edit',
  room_btn_reply: 'room_btn_reply',
  room_btn_download: 'room_btn_download',
  room_btn_copy_msg: 'room_btn_copy_msg',
  room_btn_copy_msg_link: 'room_btn_copy_msg_link',
  room_btn_copy_img: 'room_btn_copy_img',
  room_btn_share: 'room_btn_share',
  room_btn_forward: 'room_btn_forward',
  room_btn_flag: 'room_btn_flag',
  room_btn_del: 'room_btn_del',
  room_btn_clear: 'room_btn_clear',
  room_btn_invite: 'room_btn_invite',
  room_btn_pin: 'room_btn_pin',
  room_profile_name: 'room_profile_name',
  room_profile_avatar: 'room_profile_avatar',
  room_pinned_message_header: 'room_pinned_message_header',
  // RoomOptionsScreen
  options_edit_details: 'options_edit_details',
  options_btn_files: 'options_btn_files',
  options_btn_share_link: 'options_btn_share_link',
  options_btn_share_qr: 'options_btn_share_qr',
  options_btn_clear_files: 'options_btn_clear_files',
  options_btn_leave: 'options_btn_leave',
  options_btn_toggle_notifications: 'options_btn_toggle_notifications',
  options_btn_toggle_call: 'options_btn_toggle_call',
  // InviteQRScreen
  invite_btn_type: 'invite_btn_type',
  invite_btn_expires_hours: 'invite_btn_expires_hours',
  invite_btn_expires_days: 'invite_btn_expires_days',
  invite_btn_expires_weeks: 'invite_btn_expires_weeks',
  invite_btn_no_expire: 'invite_btn_no_expire',
  invite_btn_accept_admin: 'invite_btn_accept_admin',
  invite_btn_generate: 'invite_btn_generate',
  // RoomMembersScreen
  members_search: 'members_search',
  // AccountScreen
  profile_pick_avatar: 'profile_pick_avatar',
  profile_edit_name: 'profile_edit_name',
  profile_network_status: 'profile_network_status',
  account_btn_share_network_info: 'account_btn_share_network_info',
  account_btn_copy_network_info: 'account_btn_copy_network_info',
  profile_tos: 'profile_tos',
  // list
  list_item: 'list_item_' as any,
  create_channel_btn: 'create_channel_btn',
  // Avatar
  avatar_image: 'avatar_image',
  // onboarding
  onboarding_btn_skip: 'onboarding_btn_skip',
  onboarding_btn_next: 'onboarding_btn_next',
  onboarding_btn_prev: 'onboarding_btn_prev',
  onboarding_btn_finish: 'onboarding_btn_finish',
  onboarding_tos_btn_let_in: 'onboarding_tos_btn_let_in',
  onboarding_input_username: 'onboarding_input_username',
  onboarding_btn_save_username: 'onboarding_btn_save_username',
  // IdentityScreen
  identity_btn_create: 'identity_btn_create',
  identity_btn_link: 'identity_btn_link',
  identity_btn_skip: 'identity_btn_skip',
  // Room Participant list
  participant_item: 'participant_item_' as any,
  participant_btn_all: 'participant_btn_all',
  // Tooltips
  tooltip_btn_profile: 'tooltip_btn_profile',
  tooltip_btn_join_room: 'tooltip_btn_join_room',
  // PasscodeScreen
  passcode_btn_back: 'passcode_btn_back',
  passcode_btn_close: 'passcode_btn_close',
  passcode_btn_delete: 'passcode_btn_delete',
  passcode_confirm_btn_back: 'passcode_confirm_btn_back',
  passcode_confirm_btn_close: 'passcode_confirm_btn_close',
  // RecoverAccountScreen
  recover_btn_link: 'recover_btn_link',
  recover_btn_recover: 'recover_btn_recover',
  recover_input_passphrase: 'recover_input_passphrase',
  recover_btn_continue: 'recover_btn_continue',
  recover_btn_paste: 'recover_btn_paste',
  recover_btn_success: 'recover_btn_success',
  // IdentitySyncScreen
  identity_accept_checkbox: 'identity_accept_checkbox',
  identity_btn_accept: 'identity_btn_accept',
  identity_btn_use_camera_action: 'identity_btn_use_camera_action',
  identity_btn_use_invite_link: 'identity_btn_use_invite_link',
  identity_sync_invite_link_input: 'identity_sync_invite_link_input',
  identity_sync_invite_link_submit: 'identity_sync_invite_link_submit',
  identity_sync_invite_link_declined: 'identity_sync_invite_link_declined',
  identity_sync_invite_request_confirm: 'identity_sync_invite_request_confirm',
  identity_sync_invite_request_notMyDevice:
    'identity_sync_invite_request_notMyDevice',
  identity_sync_invite_request_success: 'identity_sync_invite_request_success',
  // Create ID
  create_identity_agreement_btn: 'create_identity_agreement_btn',
  create_identity_view_phrase_btn: 'create_identity_view_phrase_btn',
  create_identity_copy_phrase_btn: 'create_identity_copy_phrase_btn',
  create_identity_setup_continue_btn: 'create_identity_setup_continue_btn',
  create_id_verification_word: 'create_id_verification_word',
  create_id_submit_verification: 'create_id_submit_verification',
  create_id_onFinish_btn: 'create_id_onFinish_btn',
  // Components
  radio_btn: 'radio_btn',
  radio_btn_checked: 'radio_btn_checked',
  // Call Screen
  call_shared_screen_expand_button: 'call_shared_screen_expand_button',
  // MediaPreviewScreen
  media_preview_video_view: 'media_preview_video_view',
  media_preview_btn_custom_play: 'media_preview_btn_custom_play',
  media_preview_loading: 'media_preview_loading',
  // Discover Community
  discover_community_join_room: 'discover_community_join_room',
  discover_community_select_room: 'discover_community_select_room',

  // wallet
  wallet_default_server_action: 'wallet_default_server_action',
  wallet_custom_server_action: 'wallet_custom_server_action',
  wallet_server_get_started: 'wallet_server_get_started',
  wallet_setup_onFinish_btn: 'wallet_setup_onFinish_btn',
  wallet_create_agreement_btn: 'wallet_create_agreement_btn',
  wallet_settings_option_any: 'wallet_settings_option_any',
  wallet_settings_option_custom: 'wallet_settings_option_custom',
  wallet_settings_amount_input: 'wallet_settings_amount_input',
  wallet_settings_save_btn: 'wallet_settings_save_btn',
  wallet_send_submit_btn: 'wallet_send_submit_btn',
  wallet_send_scanner_btn: 'wallet_send_scanner_btn',
  wallet_send_scanner_back_btn: 'wallet_send_scanner_back_btn',
  wallet_send_open_scanner_btn: 'wallet_send_open_scanner_btn',
  wallet_send_paste_btn: 'wallet_send_paste_btn',
  wallet_input_wallet_address: 'wallet_input_wallet_address',
  wallet_input_transaction_description: 'wallet_input_transaction_description',
  wallet_btn_transfer_next: 'wallet_btn_transfer_next',
  wallet_btn_select_keet_contact: 'wallet_btn_select_keet_contact',
  wallet_btn_select_crypto_wallet: 'wallet_btn_select_crypto_wallet',
  wallet_receive_generate_invoice_btn: 'wallet_receive_generate_invoice_btn',
  wallet_receive_copy_invoice_btn: 'wallet_receive_copy_invoice_btn',
  wallet_receive_share_invoice_btn: 'wallet_receive_share_invoice_btn',
  wallet_custom_server_set_mode_btn: 'wallet_custom_server_set_mode_btn',
  wallet_custom_server_continue_btn: 'wallet_custom_server_continue_btn',
  wallet_custom_server_host_input: 'wallet_custom_server_host_input',
  wallet_custom_server_port_input: 'wallet_custom_server_port_input',
  wallet_custom_server_eth_indexer_input:
    'wallet_custom_server_eth_indexer_input',
  wallet_custom_server_connect_btn: 'wallet_custom_server_connect_btn',
  // User Profile Screen
  user_profile_chat: 'user_profile_chat',
  user_profile_call: 'user_profile_call',
  user_profile_send_dm: 'user_profile_send_dm',
} as const)

const DYNAMIC_KEY = [APPIUM_IDs.list_item]

export const appiumTestProps = (
  id?: keyof typeof APPIUM_IDs | string,
): Pick<ViewProps, 'accessible' | 'testID' | 'accessibilityLabel'> => {
  try {
    if (!id) {
      return {}
    }

    const appId = APPIUM_IDs[id as keyof typeof APPIUM_IDs]
    if (appId)
      return { accessible: true, testID: appId, accessibilityLabel: appId }

    if (DYNAMIC_KEY.some((substring) => _includes(id, substring))) {
      return { accessible: true, testID: id, accessibilityLabel: id }
    }
    return {}
  } catch (e) {
    console.log(e)
    return {}
  }
}
