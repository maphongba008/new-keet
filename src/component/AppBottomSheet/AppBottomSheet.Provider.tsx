import React from 'react'
import { type BottomSheetModal } from '@gorhom/bottom-sheet'
import { BackdropPressBehavior } from '@gorhom/bottom-sheet/lib/typescript/components/bottomSheetBackdrop/types'

import { ChannelRoomTipsProps } from 'screen/CreateChannel/ChannelRoomTips'

// Enum for bottom sheet types
import BottomSheetEnum from './SheetComponents/BottomSheetEnum'
// Import props/interfaces from various components
import { CallStartedInterface } from './SheetComponents/CallStarted'
import { ChatEventDeleteConfirmSheetI } from './SheetComponents/ChatEventDeleteConfirmSheet'
import { ChatEventOptionsSheetProp } from './SheetComponents/ChatEventOptionsSheet'
import { ChatLinkConfirmationInterface } from './SheetComponents/ChatLinkConfirmation'
import { ClearAllFilesInterface } from './SheetComponents/ClearAllFiles'
import { ConfirmBlockDialogInterface } from './SheetComponents/ConfirmBlockDialog'
import { ConfirmDialogInterface } from './SheetComponents/ConfirmDialog'
import { ConfirmInappropriateMessageDialogInterface } from './SheetComponents/ConfirmInappropriateMessageDialog'
import { DeviceDetailsInterface } from './SheetComponents/DeviceDetailsSheet'
import { DialogInterface } from './SheetComponents/Dialog'
import { DMRequestInterface } from './SheetComponents/DMRequest'
import { EmojiSourceSheetInterface } from './SheetComponents/EmojiSourceSheet'
import { FeedbackFormSheetInterface } from './SheetComponents/FeedbackForm'
import { IdentityBackupConfirmSheetInterface } from './SheetComponents/IdentityBackupConfirmSheet'
import { InviteTypeOptionsSheetProps } from './SheetComponents/InviteTypeOptionsSheet'
import { ListOptionSheetPropsI } from './SheetComponents/ListOptionsSheet'
import { MessageOptionsSheetInterface } from './SheetComponents/MessageOptionSheet'
import { OptionSheetProps } from './SheetComponents/OptionsSheet'
import { ParticipantsSheetProps } from './SheetComponents/ParticipantsSheet'
import { PermissionRequiredInterface } from './SheetComponents/PermissionRequired'
import { RoleChangeConfirmInterface } from './SheetComponents/RoleChangeConfirm'
import { RoomAdminResetInterface } from './SheetComponents/RoomAdminReset'
import { RoomAvatarSheetProps } from './SheetComponents/RoomAvatar'
import { RoomOptionsStatsI } from './SheetComponents/RoomOptionsStats'

// props specific for bottom sheet
interface BaseBottomSheetProps {
  bottomSheetIsTransparent?: boolean
  bottomSheetOnDismissCallback?: () => void
  bottomSheetBackdropPressBehaviour?: BackdropPressBehavior
  bottomSheetEnablePanDownToClose?: boolean
}

// props specific for bottom sheet's rendered component
type BottomSheetComponentProps = {
  [BottomSheetEnum.MessageOptionsSheet]: MessageOptionsSheetInterface
  [BottomSheetEnum.ConfirmDialog]: ConfirmDialogInterface
  [BottomSheetEnum.ConfirmBlockDialog]: ConfirmBlockDialogInterface
  [BottomSheetEnum.ConfirmInappropriateMessageDialog]: ConfirmInappropriateMessageDialogInterface
  [BottomSheetEnum.InviteTypeOptionsSheet]: InviteTypeOptionsSheetProps
  [BottomSheetEnum.RecoverWalletSheet]: {}
  [BottomSheetEnum.ParticipantsSheet]: ParticipantsSheetProps
  [BottomSheetEnum.OptionsSheet]: OptionSheetProps
  [BottomSheetEnum.CallStarted]: CallStartedInterface
  [BottomSheetEnum.RoomActionSheet]: {}
  [BottomSheetEnum.NetworkBottomSheet]: {}
  [BottomSheetEnum.ChatEventOptionsSheet]: ChatEventOptionsSheetProp
  [BottomSheetEnum.PermissionRequired]: PermissionRequiredInterface
  [BottomSheetEnum.RoomInviteType]: {}
  [BottomSheetEnum.ClearAllFiles]: ClearAllFilesInterface
  [BottomSheetEnum.ChatLinkConfirmationSheet]: ChatLinkConfirmationInterface
  [BottomSheetEnum.Dialog]: DialogInterface
  [BottomSheetEnum.FeedbackFormSheet]: FeedbackFormSheetInterface
  [BottomSheetEnum.EmojiSourceSheet]: EmojiSourceSheetInterface
  [BottomSheetEnum.RoomAvatarBottomSheet]: RoomAvatarSheetProps
  [BottomSheetEnum.ChannelRoomTips]: ChannelRoomTipsProps
  [BottomSheetEnum.IdentityBackupConfirmSheet]: IdentityBackupConfirmSheetInterface
  [BottomSheetEnum.CleanDeviceBottomSheet]: {}
  [BottomSheetEnum.TurnOffPasscodeSheet]: {}
  [BottomSheetEnum.RoomOptionsStats]: RoomOptionsStatsI
  [BottomSheetEnum.CommunityContacts]: {}
  [BottomSheetEnum.RoomAdminReset]: RoomAdminResetInterface
  [BottomSheetEnum.DeviceDetailsBottomSheet]: DeviceDetailsInterface
  [BottomSheetEnum.DMRequest]: DMRequestInterface
  [BottomSheetEnum.ChatEventDeleteConfirmSheet]: ChatEventDeleteConfirmSheetI
  [BottomSheetEnum.WalletSettings]: {}
  [BottomSheetEnum.WalletDelete]: {}
  [BottomSheetEnum.RoleChangeConfirm]: RoleChangeConfirmInterface
  [BottomSheetEnum.ListOptionsSheet]: ListOptionSheetPropsI
  [BottomSheetEnum.CallAutoEndPopup]: {}
}

type BottomSheetProps = {
  [K in keyof BottomSheetComponentProps]: {
    bottomSheetType: K
  } & BottomSheetComponentProps[K]
}[keyof BottomSheetComponentProps]

// All types are combined here so that when showing a sheet, you only need to provide the enum.
// Call showBottomSheet({ bottomSheetType: BottomSheetEnum }) and the appropriate interface/type will be inferred automatically.
export type AppSheetType = BottomSheetProps & BaseBottomSheetProps

class BottomSheetProviderClass {
  bottomSheetContext: AppSheetType
  constructor() {
    this.bottomSheetContext = {
      bottomSheetType: BottomSheetEnum.RoomActionSheet, // Default type
    }
  }
}

export const BottomSheetProvider = new BottomSheetProviderClass()

export const BottomSheetRef = React.createRef<BottomSheetModal>()
