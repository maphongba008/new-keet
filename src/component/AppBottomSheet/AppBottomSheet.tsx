import React, { memo, useCallback, useEffect, useMemo } from 'react'

import { BottomSheetProvider, BottomSheetRef } from './AppBottomSheet.Provider'
import {
  closeBottomSheet,
  showBottomSheet,
  useAppBottomSheetStore,
} from './AppBottomSheet.Store'
import SheetComponents from './SheetComponents/index'
import { BottomSheetBase } from '../BottomSheetBase'

const AppBottomSheet = () => {
  const {
    bottomSheetType,
    bottomSheetIsTransparent,
    bottomSheetBackdropPressBehaviour,
    bottomSheetEnablePanDownToClose,
    bottomSheetOnDismissCallback,
    ...rest
  } = BottomSheetProvider.bottomSheetContext

  useEffect(() => {
    const timeout = setTimeout(() => BottomSheetRef?.current?.present())
    return () => clearTimeout(timeout)
  }, [])

  const onBottomSheetClose = useCallback(() => {
    if (bottomSheetOnDismissCallback) {
      bottomSheetOnDismissCallback()
    }
    closeBottomSheet()
  }, [bottomSheetOnDismissCallback])

  const renderComponent = useMemo(() => {
    const Component = (SheetComponents[bottomSheetType] as any) || null
    return <Component {...rest} />
  }, [bottomSheetType, rest])

  return (
    <BottomSheetBase
      sheetRef={BottomSheetRef}
      isTransparent={bottomSheetIsTransparent}
      backdropPressBehaviour={bottomSheetBackdropPressBehaviour ?? 'close'}
      enablePanDownToClose={bottomSheetEnablePanDownToClose ?? true}
      onClose={onBottomSheetClose}
    >
      {renderComponent}
    </BottomSheetBase>
  )
}

export default memo(AppBottomSheet)
export { showBottomSheet, closeBottomSheet, useAppBottomSheetStore }
