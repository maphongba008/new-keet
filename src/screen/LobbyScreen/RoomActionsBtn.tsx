import React, { memo, useCallback } from 'react'
import { StyleSheet, View } from 'react-native'
import { useDispatch, useSelector } from 'react-redux'
import isEqual from 'react-fast-compare'
import LinearGradient from 'react-native-linear-gradient'

import {
  closeWalkthroughTooltip,
  getWalkthroughStep,
  getWalkthroughTooltip,
} from 'reducers/application'

import { showBottomSheet } from 'component/AppBottomSheet/AppBottomSheet'
import BottomSheetEnum from 'component/AppBottomSheet/SheetComponents/BottomSheetEnum'
import { ButtonBase } from 'component/Button'
import MaskGradient from 'component/MaskGradient'
import SvgIcon from 'component/SvgIcon'
import { createThemedStylesheet, gradient, useTheme } from 'component/theme'
import { Tooltip } from 'component/Tooltip'
import { APPIUM_IDs, appiumTestProps } from 'lib/appium'
import s, { UI_SIZE_16, UI_SIZE_18, UI_SIZE_40 } from 'lib/commonStyles'
import { setStorageWalkthroughTooltipShownDone } from 'lib/localStorage'

import { useStrings } from 'i18n/strings'

export default memo(() => {
  const strings = useStrings()
  const styles = getStyles()
  const theme = useTheme()
  const dispatch = useDispatch()
  const shouldShowTooltip = useSelector(getWalkthroughTooltip)
  const currentStep = useSelector(getWalkthroughStep)

  const onButtonBasePress = useCallback(() => {
    showBottomSheet({
      bottomSheetType: BottomSheetEnum.RoomActionSheet,
    })
  }, [])

  const onTooltipClose = useCallback(async () => {
    // Update local storage to mark the tooltip as shown
    setStorageWalkthroughTooltipShownDone()
    dispatch(closeWalkthroughTooltip())
  }, [dispatch])

  const ref = React.createRef<any>()
  return (
    <Tooltip
      componentRef={ref}
      step={2}
      content={
        <LinearGradient
          style={styles.buttonGradient}
          {...gradient.keet_gradient_brightBlue20}
        >
          <MaskGradient
            linearGradientProps={gradient.keet_gradient_brightBlue}
            MaskElement={
              <SvgIcon
                name="plus"
                width={UI_SIZE_18}
                height={UI_SIZE_18}
                color={theme.color.accent}
              />
            }
          />
        </LinearGradient>
      }
      placement="top"
      description={strings.room.walkthroughTooltip.createOrJoin}
      title={strings.room.walkthroughTooltip.getStarted}
      toShow={shouldShowTooltip && currentStep === 2}
      onClose={onTooltipClose}
      onPressNext={onTooltipClose}
      totalSteps={2}
      testProps={appiumTestProps(APPIUM_IDs.tooltip_btn_join_room)}
    >
      <View ref={ref}>
        <ButtonBase
          testID={APPIUM_IDs.lobby_btn_add_room}
          hint={strings.lobby.roomActions.createRoomBtn}
          onPress={onButtonBasePress}
        >
          <LinearGradient
            style={styles.buttonGradient}
            {...gradient.keet_gradient_brightBlue20}
          >
            <MaskGradient
              linearGradientProps={gradient.keet_gradient_brightBlue}
              MaskElement={
                <SvgIcon
                  name="plus"
                  width={UI_SIZE_18}
                  height={UI_SIZE_18}
                  color={theme.color.accent}
                />
              }
            />
          </LinearGradient>
        </ButtonBase>
      </View>
    </Tooltip>
  )
}, isEqual)

const getStyles = createThemedStylesheet((theme) => {
  const styles = StyleSheet.create({
    buttonGradient: {
      ...s.centeredLayout,
      borderRadius: theme.border.radiusNormal,
      height: UI_SIZE_40,
      width: UI_SIZE_40,
    },
    tooltipContainer: {
      backgroundColor: theme.color.bg,
      borderRadius: theme.border.radiusNormal,
      padding: UI_SIZE_16,
    },
  })
  return styles
})
