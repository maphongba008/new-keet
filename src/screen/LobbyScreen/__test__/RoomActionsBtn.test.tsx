import { fireEvent, screen } from '@testing-library/react-native'

import BottomSheetEnum from 'component/AppBottomSheet/SheetComponents/BottomSheetEnum'
import { APPIUM_IDs } from 'lib/appium'
import { renderWithProviders } from 'lib/testUtils'

import * as SheetFn from '../../../component/AppBottomSheet/AppBottomSheet.Store'
import RoomActionsBtn from '../RoomActionsBtn'

jest.spyOn(SheetFn, 'showBottomSheet').mockReturnValue(undefined)

describe('RoomListFilter', () => {
  test('render correctly', async () => {
    const element = <RoomActionsBtn />
    const result = renderWithProviders(element)
    expect(result.toJSON()).toMatchSnapshot()
    // check if showBottomSheet was called when button Add is pressed
    const btn = await screen.findByTestId(APPIUM_IDs.lobby_btn_add_room)
    fireEvent.press(btn)
    expect(SheetFn.showBottomSheet).toHaveBeenCalledWith({
      bottomSheetType: BottomSheetEnum.RoomActionSheet,
    })
  })
})
