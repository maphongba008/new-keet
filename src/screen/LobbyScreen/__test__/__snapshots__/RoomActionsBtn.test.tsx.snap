// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomListFilter render correctly 1`] = `
<View
  onLayout={[Function]}
>
  <View
    accessibilityHint="Create Room"
    accessibilityLabel="lobby_btn_add_room"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "opacity": 1,
        },
        undefined,
      ]
    }
    testID="lobby_btn_add_room"
  >
    <BVLinearGradient
      colors={
        [
          855703503,
          860530431,
        ]
      }
      endPoint={
        {
          "x": 0,
          "y": 1,
        }
      }
      locations={null}
      startPoint={
        {
          "x": 0,
          "y": 0,
        }
      }
      style={
        {
          "alignItems": "center",
          "borderRadius": 8,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
        }
      }
    >
      <RNCMaskedView>
        <View
          pointerEvents="none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={18}
            bbWidth={18}
            color="#95E6CB"
            fill="#95E6CB"
            focusable={false}
            height={18}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {},
                {
                  "flex": 0,
                  "height": 18,
                  "width": 18,
                },
              ]
            }
            testID="svg-icon-plus"
            tintColor="#95E6CB"
            vbHeight={24}
            vbWidth={24}
            width={18}
            xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M20.813 12a.56.56 0 0 1-.563.563h-7.687v7.687a.562.562 0 1 1-1.126 0v-7.687H3.75a.563.563 0 0 1 0-1.126h7.688V3.75a.563.563 0 0 1 1.124 0v7.688h7.688a.56.56 0 0 1 .563.562" fill="#fff"/></svg>
"
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4288014027,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M20.813 12a.56.56 0 0 1-.563.563h-7.687v7.687a.562.562 0 1 1-1.126 0v-7.687H3.75a.563.563 0 0 1 0-1.126h7.688V3.75a.563.563 0 0 1 1.124 0v7.688h7.688a.56.56 0 0 1 .563.562"
                fill={
                  {
                    "payload": 4294967295,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={18}
          bbWidth={18}
          color="#95E6CB"
          fill="#95E6CB"
          focusable={false}
          height={18}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {},
              {
                "flex": 0,
                "height": 18,
                "width": 18,
              },
            ]
          }
          testID="svg-icon-plus"
          tintColor="#95E6CB"
          vbHeight={24}
          vbWidth={24}
          width={18}
          xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M20.813 12a.56.56 0 0 1-.563.563h-7.687v7.687a.562.562 0 1 1-1.126 0v-7.687H3.75a.563.563 0 0 1 0-1.126h7.688V3.75a.563.563 0 0 1 1.124 0v7.688h7.688a.56.56 0 0 1 .563.562" fill="#fff"/></svg>
"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4288014027,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M20.813 12a.56.56 0 0 1-.563.563h-7.687v7.687a.562.562 0 1 1-1.126 0v-7.687H3.75a.563.563 0 0 1 0-1.126h7.688V3.75a.563.563 0 0 1 1.124 0v7.688h7.688a.56.56 0 0 1 .563.562"
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
        <BVLinearGradient
          colors={
            [
              4278255567,
              4283082495,
            ]
          }
          endPoint={
            {
              "x": 0,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            {
              "x": 0,
              "y": 0,
            }
          }
          style={
            {
              "flex": 1,
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            }
          }
        />
      </RNCMaskedView>
    </BVLinearGradient>
  </View>
</View>
`;
